### PR TITLE
Add toolchain_type and toolchain definitions for Android SDK.

### PR DESCRIPTION
**Pull request by katre**

### CODE REVIEW

File: tools/android/BUILD.tools [hunk 1]
```diff
@@ -3,6 +3,9 @@ load("//tools/python:private/version.bzl", "PY_BINARY_VERSION")
 
 package(default_visibility = ["//visibility:public"])
 
+# The toolchain type used to distinguish Android SDK toolchains.
+toolchain_type(name = "sdk_toolchain_type")
```
Comment (juliexxia) [inline tools/android/BUILD.tools:7]: Why not use the regular toolchain_type naming convention? Maybe worth leaving a note explaining so this isn't copied without understanding? 

Is it because this file isn't clearly for just sdk logic so the toolchain type purpose isn't obvious from the target path? That would make sense to me. i guess probably not worth creating an sdk subdir for it?

File: tools/android/android_sdk_repository_template.bzl [hunk 2]
```diff
@@ -159,6 +160,16 @@ def create_android_sdk_rules(
...
         )
 
+        native.toolchain(
+            name = "sdk-%d-toolchain" % api_level,
+            toolchain_type = "@bazel_tools//tools/android:sdk_toolchain_type",
+            exec_compatible_with = HOST_CONSTRAINTS,
+            target_compatible_with = [
+                "@bazel_tools//platforms:android",
+            ],
+            toolchain = ":sdk-%d" % api_level,
```
Comment (juliexxia) [inline tools/android/android_sdk_repository_template.bzl:170]: Will the toolchains also live in that same BUILD.tools ?

File: tools/android/BUILD.tools (hunk 1 above)
Comment (katre) [reply tools/android/BUILD.tools:7]: I wanted to clarify that this is specifically for the SDK. I don't think we'll need an NDK toolchain type (that's really just a special case of the C++ toolchain type), but there's also various other things in Android that might be toolchains.

File: tools/android/android_sdk_repository_template.bzl (hunk 2 above)
Comment (katre) [reply tools/android/android_sdk_repository_template.bzl:170]: No, this file is loaded from the BUILD file written by `android_sdk_repository`, so the toolchains are part of that repo.

File: src/test/java/com/google/devtools/build/lib/bazel/rules/android/AndroidSdkRepositoryTest.java [hunk 3]
```diff
@@ -38,6 +38,7 @@ public void setup() throws Exception {
     scratch.file(
         "bazel_tools_workspace/tools/android/android_sdk_repository_template.bzl",
         ResourceLoader.readFromResources("tools/android/android_sdk_repository_template.bzl"));
+    scratch.appendFile("WORKSPACE", "local_config_platform(name='local_config_platform')");
```
Comment (juliexxia) [inline src/test/java/com/google/devtools/build/lib/bazel/rules/android/AndroidSdkRepositoryTest.java:41]: Why is this change needed?

File: tools/android/BUILD.tools (hunk 1 above)
Comment (juliexxia) [reply tools/android/BUILD.tools:7]: Yeah that makes sense to me. I'd say maybe leave a comment that this isn't a conventional toolchain type name (re: https://docs.bazel.build/versions/master/toolchains.html#writing-rules-that-use-toolchains) but up to you

Comment (katre) [reply tools/android/BUILD.tools:7]: I don't think the comment is helpful, there are plenty of other examples to follow.

File: src/test/java/com/google/devtools/build/lib/bazel/rules/android/AndroidSdkRepositoryTest.java (hunk 3 above)
Comment (katre) [reply src/test/java/com/google/devtools/build/lib/bazel/rules/android/AndroidSdkRepositoryTest.java:41]: The androidsdk repo reads constraints.bzl from `@local_config_platform` (to determine the host constraints for the `exec_compatible_with`), so it needs to be defined.
